drivers/at86rf2xx/include/at86rf2xx_aes.h

```diff
@@ -0,0 +1,259 @@
+ */
+#define AT86RF2XX_AES_KEY_LENGTH                        ((AT86RF2XX_AES_KEY_BITS) / 8)
+/**
+ * @brief   AES block size in bytes
+ */
+#define AT86RF2XX_AES_BLOCK_SIZE                        ((AT86RF2XX_AES_KEY_BITS) / 8)
+/**
+ * @brief   Time to complete the AES algorithm in us
+ */
+#define AT86RF2XX_AES_DELAY_US                          (24)
+
+/**
+ * @name    AES rigister addresses
+ * @{
+ */
+#define AT86RF2XX_REG__AES_STATUS                       (0x82)
+#define AT86RF2XX_REG__AES_CTRL                         (0x83)
+#define AT86RF2XX_REG__AES_KEY_START                    (0x84)
+#define AT86RF2XX_REG__AES_KEY_END                      (0x93)
+#define AT86RF2XX_REG__AES_DATA_START                   (0x84)
+#define AT86RF2XX_REG__AES_DATA_END                     (0x93)
+#define AT86RF2XX_REG__AES_CTRL_MIRROR                  (0x94)
+/** @} */
+
+/**
+ * @name    Layout of register AES_STATUS
+ * @{
+ */
+#define AT86RF2XX_AES_STATUS_MASK__AES_ER               (0x80)
+#define AT86RF2XX_AES_STATUS_MASK__AES_DONE             (0x01)
+
+#define AT86RF2XX_AES_STATUS_AES_ER__NO_ERROR           (0x00)
+#define AT86RF2XX_AES_STATUS_AES_ER__ERROR              (0x80)
+
+#define AT86RF2XX_AES_STATUS_AES_DONE__NOT_DONE         (0x00)
+#define AT86RF2XX_AES_STATUS_AES_DONE__DONE             (0x01)
+/** @} */
+
+/**
+ * @name    Layout of register AES_CTRL
+ * @{
+ */
+#define AT86RF2XX_AES_CTRL_MASK__AES_REQUEST            (0x80)
+#define AT86RF2XX_AES_CTRL_MASK__AES_MODE               (0x70)
+#define AT86RF2XX_AES_CTRL_MASK__AES_DIR                (0x08)
+
+#define AT86RF2XX_AES_CTRL_AES_REQUEST__IDLE            (0x00)
+#define AT86RF2XX_AES_CTRL_AES_REQUEST__START           (0x80)
+
+#define AT86RF2XX_AES_CTRL_AES_MODE__ECB                (0x00)
+#define AT86RF2XX_AES_CTRL_AES_MODE__KEY                (0x10)
+#define AT86RF2XX_AES_CTRL_AES_MODE__CBC                (0x20)
+
+#define AT86RF2XX_AES_CTRL_AES_DIR__ENC                 (0x00)
+#define AT86RF2XX_AES_CTRL_AES_DIR__DEC                 (0x08)
+/** @} */
+
+/**
+ * @name    Layout of register AES_CTRL_MIRROR
+ * @{
+ */
+#define AT86RF2XX_AES_CTRL_MIRROR_MASK__AES_REQUEST     (0x80)
+#define AT86RF2XX_AES_CTRL_MIRROR_MASK__AES_MODE        (0x70)
+#define AT86RF2XX_AES_CTRL_MIRROR_MASK__AES_DIR         (0x08)
+
+#define AT86RF2XX_AES_CTRL_MIRROR_AES_REQUEST__IDLE     (0x00)
+#define AT86RF2XX_AES_CTRL_MIRROR_AES_REQUEST__START    (0x80)
+
+#define AT86RF2XX_AES_CTRL_MIRROR_AES_MODE__ECB         (0x00)
+#define AT86RF2XX_AES_CTRL_MIRROR_AES_MODE__KEY         (0x10)
+#define AT86RF2XX_AES_CTRL_MIRROR_AES_MODE__CBC         (0x20)
+
+#define AT86RF2XX_AES_CTRL_MIRROR_AES_DIR__ENC          (0x00)
+#define AT86RF2XX_AES_CTRL_MIRROR_AES_DIR__DEC          (0x08)
+/** @} */
+
+/**
+ * @brief   An AES block
+ *
+ * AES works on blocks of 16 bytes
+ */
+typedef uint8_t aes_block_t[AT86RF2XX_AES_BLOCK_SIZE];
+
+/**
+ * @brief   Pointer to @ref aes_block_t
+ */
+typedef aes_block_t* aes_block_ptr_t;
```

Comment:
I'd drop that, it obscures the pointer nature of the variables which is just confusing.
Nothing wrong about just using `aes_block_t*`

Reply:
Ok